lib/ews/soap/exchange_web_service.rb

```diff
@@ -219,11 +219,9 @@ def parse_soap_response(soapmsg, opts = {})
       raise EwsError, "Can't parse an empty response. Please check your endpoint." if(soapmsg.nil?)
       opts[:response_class] ||= EwsSoapResponse
       EwsParser.new(soapmsg).parse(opts)
-    rescue Exception => e
-      @log.error "Error parsing soap response:"
-      @log.error soapmsg
-      @log.error e
-      raise e
+    rescue
+      @log.warn "Error parsing soap response: #{soapmsg}"
+      raise
```

Comment:
As far as I can tell, `soapmsg` is [this](https://kibana.outreach-internal.com/app/kibana#/doc/logstash-*/logstash-datalog-2017.10.18/outreach-app?id=AV8wwQ6uABfRwJGnFdf7&_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:now-24h,mode:quick,to:now))), which seems to include the message body.

It may be nice to log the error instead, as it includes the backtrace and `error.class`, and omit `soapmsg` altogether.

Reply:
I don't think so. The one you give is for "Viewpoint::EWS::EwsParseError: Error parsing SAX document: ...", see here https://kibana.outreach-internal.com/goto/75b7cb2dc7014feae6b9c86e351fa797

This PR is trying to solve "undefined method `[]' for nil:NilClass", like this https://kibana.outreach-internal.com/goto/c27ebd25287541c83636929118c43fbf

The `soapmsg` is the original xml response payload, it should be safe to include inside our log. As the error is being re-raised, our outside component will actually rescue and log with all proper backtraces. The only thing here that might be helpful is to record the `soapmsg` in the log, as it might help diagnosis - if not for this purpose, I might just completely remove this `rescue...log...raise` block.